cmd/sansshell-server/server/server.go

```diff
@@ -192,3 +198,22 @@ func Run(ctx context.Context, opts ...Option) {
 		os.Exit(1)
 	}
 }
+
+// extracts transport credentials from runState. Will error if both credSource and tlsConfig are both specified
+func extractTransportCredentialsFromRunState(ctx context.Context, rs *runState) credentials.TransportCredentials {
+	var creds credentials.TransportCredentials
+	var err error
+	if rs.credSource != "" && rs.tlsConfig != nil {
+		rs.logger.Error(nil, "both credSource and tlsConfig are defined", "credSource", rs.credSource)
+		os.Exit(1)
+	} else if rs.credSource != "" {
+		creds, err = mtls.LoadServerCredentials(ctx, rs.credSource)
+		if err != nil {
+			rs.logger.Error(err, "mtls.LoadServerCredentials", "credsource", rs.credSource)
+			os.Exit(1)
```

Comment:
Make this function simply return an error and then have Run() log it and exit. That way all exit handling is in one place.

Reply:
ok, thanks! I was not sure what the best practices were for error handling like that.